extension.js

```diff
@@ -313,9 +317,50 @@ function workarea_for_window(win) {
 }
 
 function target_rect_for_workarea(workarea) {
-    const target_rect = workarea.copy();
-    target_rect.height *= settings.get_double('window-height');
-    return target_rect;
+    const available_width = workarea.width,
+        available_height = workarea.height,
+        full_width = available_width + workarea.x,
+        full_height =  available_height + workarea.y,
```

Comment:
What does `full_width` and `full_height` mean? I. e. "full height" of what?

Reply:
Full height of the monitor. x and y are working as kinda offset. so full-height is available height + offset.

Reply:
What you're calculating here is not a full height of the monitor. At least if there is a bottom panel. And the same for `full_width` - if the user enables an extension that adds a panel on the right, it won't match the width of the monitor.